### PR TITLE
Add `includeInternal` to C# NodeExtensions and avoid printing errors in `GetChildOrNull`

### LINKED ISSUE #57128: GetChildOrNull<T>(int idx) thows exception if index is out of bounds
### Godot version

4.0 alpha1

### System information

Windows 10

### Issue description

`GetNodeOrNull<T>(string path)` returns `null` if the path is invalid. `GetChildOrNull<T>(int idx)` instead throws an index out of bounds exception if the index is incorrect. I feel this is an oversight, and should probably behave similarly.

### Steps to reproduce

N/A

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by raulsntos**
- Node methods in C# extended to use generics now have the optional parameter `includeInternal` like their non-generic equivalents (added to core in #30391). Also fixes a typo in the `Node.xml` documentation.
- Avoid printing an error in `GetChildOrNull` when the given index is out of range, similar to how the LINQ `ElementAtOrDefault` method works (fixes #57128).

### CODE REVIEW

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Extensions/NodeExtensions.cs [hunk 1]
```diff
@@ -93,18 +93,22 @@ public partial class Node
         /// Negative indices access the children from the last one.
         /// To access a child node via its name, use <see cref="GetNode"/>.
         /// </summary>
-        /// <seealso cref="GetChildOrNull{T}(int)"/>
+        /// <seealso cref="GetChildOrNull{T}(int, bool)"/>
         /// <param name="idx">Child index.</param>
+        /// <param name="includeInternal">
+        /// If <see langword="true"/>, internal children are skipped (see <c>internal</c>
```
Comment (kleonc) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Extensions/NodeExtensions.cs:99]: Should be _"if `false`"_. I see it's a copy-paste so would need to be fixed in `Node.xml` as well.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Extensions/NodeExtensions.cs [hunk 2]
```diff
@@ -113,15 +117,20 @@ public partial class Node
...
         /// <typeparam name="T">The type to cast to. Should be a descendant of <see cref="Node"/>.</typeparam>
         /// <returns>
         /// The child <see cref="Node"/> at the given index <paramref name="idx"/>, or <see langword="null"/> if not found.
         /// </returns>
-        public T GetChildOrNull<T>(int idx) where T : class
+        public T GetChildOrNull<T>(int idx, bool includeInternal = false) where T : class
         {
-            return GetChild(idx) as T;
+            int count = GetChildCount();
+            return idx > -count && idx < count ? GetChild(idx, includeInternal) as T : null;
```
Comment (kleonc) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Extensions/NodeExtensions.cs:133]: `-count` is the first element index, `-1` last.
```suggestion
            return idx >= -count && idx < count ? GetChild(idx, includeInternal) as T : null;
```

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Extensions/NodeExtensions.cs [hunk 3]
```diff
@@ -113,15 +117,20 @@ public partial class Node
...
+        /// </param>
         /// <typeparam name="T">The type to cast to. Should be a descendant of <see cref="Node"/>.</typeparam>
         /// <returns>
         /// The child <see cref="Node"/> at the given index <paramref name="idx"/>, or <see langword="null"/> if not found.
         /// </returns>
-        public T GetChildOrNull<T>(int idx) where T : class
+        public T GetChildOrNull<T>(int idx, bool includeInternal = false) where T : class
         {
-            return GetChild(idx) as T;
+            int count = GetChildCount();
```
Comment (kleonc) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Extensions/NodeExtensions.cs:132]:
```suggestion
            int count = GetChildCount(includeInternal);
```

**Review by kleonc — APPROVED** [review]

**Comment by neikeq** [comment]
Where was the exception being thrown?

**Comment by raulsntos** [comment]
> Where was the exception being thrown?

Sorry, not an exception. It's just an error printed by Godot C++ here:

https://github.com/godotengine/godot/blob/792f7cc28c346a06564f6da54520cb02a837f04d/scene/main/node.cpp#L1250

**Comment by neikeq** [comment]
Ah, I see. Makes sense then. Can you change the commit message to make it clear that what it avoids is printing an error?

**Review by neikeq — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**